hikari/channels.py

```diff
@@ -1339,3 +1348,22 @@ class GuildStageChannel(GuildChannel):
 
     If this is `0`, then assume no limit.
     """
+
+
+WebhookChannelT = typing.Union[GuildTextChannel, GuildNewsChannel]
+"""Union of the channel types which incoming and follower webhooks can be attached to.
+
+The following types are in this:
+
+* `GuildTextChannel`
+* `GuildNewsChannel`
+"""
+
+WebhookChannelTypes = (GuildTextChannel, GuildNewsChannel)
+"""Tuple of the channel types which are valid for `WebhookChannelT`.
+
+This includes:
+
+* `GuildTextChannel`
+* `GuildNewsChannel`
+"""
```

Comment:
Are these really needed?

Reply:
Well the whole WebhookChannelT specific typing isn't necessary so WebhookChannelT doesn't need to exist (could just document all that stuff as GuildTextableChannel and ignore the fact that some of the types which will be covered by that won't exist in the future) and WebhookChannelTypes exists for cases where someone would want to use an instance check to narrow down a GuildTextableChannel to a WebhookChannelT so it's about as unnecessary as that.

Reply:
Also, aren't these both just `TextableGuildChannel`?

Reply:
No because one includes thread channels and the other doesn't, this is covered the pr's description

Reply:
Considering it is only two types, I would lend towards specifying both instead of having a random type which pollutes the root package

Reply:
which then leads to cases where the types on specific values come out of sync with the actual values as stuff is added but only some of the methods which refer to that specific union are updated

Reply:
Evened it out a bit by removing some top level types which were just `Union[ActualType, str/int]`

Reply:
Could the be fetched by using the following to replace them? 

```py
import typing


assert isinstance(TextChannel(...), typing.get_args(WebhookChannelsT)) # True
# or (without typing module)
assert isinstance(TextChannel(...), WebhookChannelsT.__args__) # True, but __args__ is undocumented
```

Reply:
Since typing.get_args is typed as returning `Tuple[Any, ...]` that would do absolutely nothing to narrow the scope of a variable with type checkers

Reply:
That Tuple can be used for isinstances, which is what this variable is now used for

Reply:
which won't be compatible with type checkers